src/Serializer/SnippetDenormalizer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace eLife\ApiSdk\Serializer;
+
+use eLife\ApiSdk\Promise\CallbackPromise;
+use GuzzleHttp\Promise\PromiseInterface;
+
+final class SnippetDenormalizer
+{
+    private $id;
+    private $fetcher;
+    private $identityMap;
+    private $globalCallback;
+
+    public function __construct(callable $id, callable $fetcher)
+    {
+        $this->id = $id;
+        $this->fetcher = $fetcher;
+        $this->identityMap = new IdentityMap();
+    }
+
+    public function denormalizeSnippet(array $item) : PromiseInterface
+    {
+        $id = call_user_func($this->id, $item);
+
+        if ($this->identityMap->has($id)) {
```

Comment:
this isn't a form of cache because the IdentityMap gets fully reset  on the globalCallback firing? If so, is it necessary at all?